build/root.build

```diff
@@ -8,3 +8,18 @@ cxx{*}: extension = cc
 # The test target for cross-testing (running tests under Wine, etc).
 #
 test.target = $cxx.target
+
+# Disable support for building shared libs on windows with msvc
+# Upstream does not support this
+# Public headers are not marked appropriately with declspec(dllimport) and declspec(dllexport)
+# Causes severe problems with linking due to static variables and inline functions in headers
+if ($cxx.target.system == 'win32-msvc')
+{
+  switch $config.bin.lib
```

Comment:
I think this should be only `$bin.lib`.

Reply:
As I understand it, locally in a `buildfile`, eg. `$config.bin.lib` is the value passed from the build driver, and `$bin.lib` is the value for _this package_. The former is immutable, the latter is mutable. In this case, it probably doesn't matter, but it could be that `$config.bin.lib != $bin.lib` (in case somewhere in-between invoking the build driver & reaching `root.build` the latter value is modified (similar to what we are doing here).
Probably nit-picking, but just `$bin.lib` seems more correct.

Reply:
Yes, I also think `$config.bin.lib` is a value passed from the build driver and it immutable by packages. If this assumption is true, then wouldn't checking `$config.bin.lib` make more sense, since it is immutable, so no other package can change it. It also then capture's the users actual intent. `$bin.lib` is `google-benchmark`'s local "copy" that can only be changed by the package's buildfiles. @boris-kolpackov could you shed some light on the differences between `$config.bin.lib` and `$bin.lib`.
The change itself is ok to do. Just want to learn how these variables interact.

Reply:
Yeah true, though _if_ `config.bin.lib` would differ to `bin.lib` (for whatever reason), this check could then do the "error out" decision incorrectly.
But as you say, this scenario might not even be possible to achieve.

Reply:
Project-local `bin.lib` is initialized with a value from global `config.bin.lib` (or with a target platform-appropriate default). So checking/modifying `bin.lib` is the correct way to do it.